contracts/cities/nyc/local/newyorkcitycoin-token-v2.clar

```diff
@@ -0,0 +1,222 @@
+;; NEWYORKCITYCOIN TOKEN V2 CONTRACT
+;; CityCoins Protocol Version 2.0.0
+
+;; TRAIT DEFINITIONS
+
+(impl-trait .citycoin-token-trait.citycoin-token)
+(use-trait coreTrait .citycoin-core-trait.citycoin-core)
+
+;; ERROR CODES
+
+(define-constant ERR_UNAUTHORIZED (err u2000))
+(define-constant ERR_TOKEN_NOT_ACTIVATED (err u2001))
+(define-constant ERR_TOKEN_ALREADY_ACTIVATED (err u2002))
+(define-constant ERR_V1_BALANCE_NOT_FOUND (err u2003))
+
+;; SIP-010 DEFINITION
+
+(impl-trait .sip-010-trait-ft-standard.sip-010-trait)
+;; MAINNET
+;; (impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-010-trait-ft-standard.sip-010-trait)
+
+(define-fungible-token newyorkcitycoin)
+
+(define-constant DECIMALS u6)
+(define-constant MICRO_CITYCOINS (pow u10 DECIMALS))
+
+;; SIP-010 FUNCTIONS
+
+(define-public (transfer (amount uint) (from principal) (to principal) (memo (optional (buff 34))))
+  (begin
+    (asserts! (is-eq from tx-sender) ERR_UNAUTHORIZED)
+    (if (is-some memo)
+      (print memo)
+      none
+    )
+    (ft-transfer? newyorkcitycoin amount from to)
+  )
+)
+
+(define-read-only (get-name)
+  (ok "newyorkcitycoin")
+)
+
+(define-read-only (get-symbol)
+  (ok "NYC")
+)
+
+(define-read-only (get-decimals)
+  (ok DECIMALS)
+)
+
+(define-read-only (get-balance (user principal))
+  (ok (ft-get-balance newyorkcitycoin user))
+)
+
+(define-read-only (get-total-supply)
+  (ok (ft-get-supply newyorkcitycoin))
+)
+
+(define-read-only (get-token-uri)
+  (ok (var-get tokenUri))
+)
+
+;; TOKEN CONFIGURATION
+
+;; define bonus period and initial epoch length
+(define-constant TOKEN_BONUS_PERIOD u10000)
+(define-constant TOKEN_EPOCH_LENGTH u25000)
+
+;; store block height at each halving, set by register-user in core contract 
+(define-data-var coinbaseThreshold1 uint u0)
+(define-data-var coinbaseThreshold2 uint u0)
+(define-data-var coinbaseThreshold3 uint u0)
+(define-data-var coinbaseThreshold4 uint u0)
+(define-data-var coinbaseThreshold5 uint u0)
+
+;; once activated, thresholds cannot be updated again
+(define-data-var tokenActivated bool false)
+
+;; core contract states
+(define-constant STATE_DEPLOYED u0)
+(define-constant STATE_ACTIVE u1)
+(define-constant STATE_INACTIVE u2)
+
+;; one-time function to activate the token
+(define-public (activate-token (coreContract principal) (stacksHeight uint))
+  (let
+    (
+      (coreContractMap (try! (contract-call? .newyorkcitycoin-auth get-core-contract-info coreContract)))
```

Comment:
I think here v2 should be called.

Reply:
fixed in https://github.com/citycoins/contracts/pull/190/commits/f18d325c5d6effec0e70a2fba386f401eeb8e74e